### ISSUE TITLE
Upload blob

### ISSUE THREAD

**Issue by odesey** [Lepozepo/cloudinary#102]
I am trying to extract some images from `.epub` files and then save them to Cloudinary but I am not having any luck. Here is what I've done thus far:

Using `epub.js` I load the `.epub` file and read the image data:

```javascript

 const book = ePub('path_to_file.epub');
 const cover = book.coverUrl();
/* cover._result returns this:
"blob:http://localhost:3000/6433bd4e-13dc-462b-b7be-e9654ad06c18"
*/
```
I can manually assign `cover._url` to an image source and the image renders as expected:
```javascript
  $('#bookImage').attr('src', cover._result)
```
I have tried to upload the `cover._result` , `"blob:http://localhost:3000/6433bd4e-13dc-462b-b7be-e9654ad06c18"` like so:

```javascript
          Cloudinary.upload( cover._result,{
            folder: 'Books',
            resource_type: 'raw',
          },function(err, res){ 
            if (err){ 
              console.log(err); 
            } else { 
              console.log(res);
               } 
             });
```
and I get no response

I also tried this:
```javascript
const myFile = new File([cover._result], "new_file.jpg", {type: "image/jpg"}); 
Cloudinary.upload( myFile,{...})
```
The file uploads, BUT its garbage data and not readable

Lastly I tried converting the file to base64 like so:

```javascript

 var toDataUrl = function(cover, callback) {
  var xhr = new XMLHttpRequest();
  xhr.responseType = 'blob';
  xhr.onload = function() {
    var reader = new FileReader();
    reader.onloadend = function() {
      callback(reader.result);
    }
    reader.readAsDataURL(xhr.response);
  };
  xhr.open('GET', cover);
  xhr.send();
}

/******************/
      toDataUrl (cover._result, function( base64Img ) {
        Cloudinary.upload( base64Img,{
          folder: 'Books',
           resource_type: 'raw',
         },function(err, res){ 
            if (err){ 
              console.log(err); 
            } else { 
              console.log(res);
            } 
        });
    })
```
I get no response from the `Cloudinary.upload` function.

So my question, is this even possible with the package? If so what am I missing?

Thanks.

**Comment by odesey** [comment]
Solved this based on info found here #19 

Changed this:

```javascript

      toDataUrl (cover._result, function( base64Img ) {
        Cloudinary.upload( base64Img,{
          folder: 'Books',
           resource_type: 'raw',
         },function(err, res){ 
            if (err){ 
              console.log(err); 
            } else { 
              console.log(res);
            } 
        });
    })
```

to this:

```javascript

      toDataUrl (cover._result, function( base64Img ) {
        Cloudinary._upload_file( base64Img,{
          folder: 'Books',
         },function(err, res){ 
            if (err){ 
              console.log(err); 
            } else { 
              console.log(res);
            } 
        });
    })
```
Problem solved. 

This should probably be in the documentation.

**Comment by Lepozepo** [comment]
Awesome! Glad you solved it, I'll add it to the docs.
